crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -0,0 +1,328 @@
+                }),
+            _ => Ok(Self {
+                card_transaction_type,
+                transaction_id,
+                amount: None,
+            }),
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Card {
+    card_number: Secret<String, pii::CardNumber>,
+    expiration_month: String,
+    expiration_year: String,
+    security_code: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CardHolderInfo {
+    first_name: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BluesnapPaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let capture_method = if let Some(capture_method) = item.request.capture_method {
+                    capture_method.to_string()
+                } else {
+                    "automatic".to_string()
+                };
+                let auth_mode = if capture_method.to_lowercase() == "manual" {
+                    BluesnapPaymentStatus::AuthOnly
+                } else {
+                    BluesnapPaymentStatus::AuthCapture
+                };
+                let payment_request = Self {
+                    amount: item.request.amount.to_string(),
+                    credit_card: Some(Card {
+                        card_number: ccard.card_number.clone(),
+                        expiration_month: ccard.card_exp_month.peek().clone(),
+                        expiration_year: ccard.card_exp_year.peek().clone(),
+                        security_code: ccard.card_cvc.peek().clone(),
+                    }),
+                    currency: item.request.currency.to_string(),
+                    wallet_id: None,
+                    soft_descriptor: item.description.clone(),
+                    card_transaction_type: auth_mode,
+                    card_holder_info: Some(CardHolderInfo {
+                        first_name: ccard.card_holder_name.peek().clone(),
+                    }),
+                };
+                Ok(payment_request)
+            }
+            api::PaymentMethod::Wallet(ref _wallet_data) => {
+                let capture_method = if let Some(capture_method) = item.request.capture_method {
+                    capture_method.to_string()
+                } else {
+                    "automatic".to_string()
+                };
+                let auth_mode = if capture_method.to_lowercase() == "manual" {
+                    BluesnapPaymentStatus::AuthOnly
+                } else {
+                    BluesnapPaymentStatus::AuthCapture
+                };
+                let payment_request = Self {
+                    wallet_id: Some("21".to_string()),
```

Comment:
No need to write `payment_request` again, you can see other connectors to understand how to use the same struct with different values.